syntax = "proto3";

package defl;

enum RequestMethod {
  FETCH_W_LAST = 0;
  NEW_WEIGHTS = 1;
  NEW_EPOCH = 2;
  // Registration
  CLIENT_REGISTER = 7;
}

message SocketInfo {
  string host = 1;
  int32 port = 2;
}

message ClientRequest {
  RequestMethod method = 1;
  string request_uuid = 2;
  string client_name = 3;
  optional SocketInfo socket = 4;
  optional int64 target_epoch_id = 5;
  optional bytes weights = 6;
}

message Response {
  string msg = 1;
  map<string, bytes> w_last = 2;
}
